src/main/scala/monix/nio/tcp/AsyncTcpClientObservable.scala

```diff
@@ -0,0 +1,135 @@
+package monix.nio.tcp
+
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+
+import monix.eval.{Callback, Task}
+import monix.execution.Ack.{Continue, Stop}
+import monix.execution.atomic.Atomic
+import monix.execution.{Cancelable, UncaughtExceptionReporter}
+import monix.execution.cancelables.SingleAssignmentCancelable
+import monix.execution.exceptions.APIContractViolationException
+import monix.nio.{Bytes, EmptyBytes, NonEmptyBytes}
+import monix.nio.cancelables.SingleFunctionCallCancelable
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.Promise
+import scala.util.control.NonFatal
+
+class AsyncTcpClientObservable private[tcp] (
```

Comment:
As with the Consumer, there are quite some differences for TCP: waiting for connected state, we don't have a current `position`.
The similarities are that the both use the monix reactive traits. Too much abstraction and indirection for the sake of it this early is uncalled for in my opinion